[1/1] Add docs for EAS Insights

christopherwalter opened the pull request:
# Why
Because we're launching tomorrow!

# How
Copy the structure of `docs/eas-update` and fill in an introduction page for EAS Insights.

# Test Plan
Can I run a local server to preview these changes?

christopherwalter committed:
[1/1] Add docs for EAS Insights

# Why
Because we're launching tomorrow!

# How
Copy the structure of `docs/eas-update` and fill in an introduction page
for EAS Insights.

# Test Plan
Spellcheck

brentvatne commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,33 @@
+description: An introduction to EAS Insights which is a preview service for projects using the expo-insights library.
+---
+
+**EAS Insights** is a service that offers a view into a project's performance,
```
no need limit line length, let the editor do the wrapping for you

brentvatne replied:
```suggestion
**EAS Insights** is a service that will a view into a project's performance,
```

brentvatne commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,33 @@
+
+### Integration with EAS Update
+
+For those using `expo-updates` with EAS Update, we're able to provide certain
```
```suggestion
For those already using [EAS Update](/eas-update/introduction.mdx), we're able to provide certain
```

brentvatne commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,33 @@
+
+For those using `expo-updates` with EAS Update, we're able to provide certain
+high-level usage insights without any additional client-side changes. This is
+possible due to the nature of EAS Update + expo-updates, where clients check for
+available updates and fetch them if possible. Data about these client requests
+is aggregated into a limited Insights view that shows usage over time, as well
+as usage broken down by platform.
```
```suggestion
possible because by aggregating data from client requests to check for an update into a limited Insights view that shows usage over time, as well
as usage broken down by platform.
```

brentvatne replied:
(you can see here one of the downsides of manually wrapping docs)

brentvatne commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,33 @@
+is aggregated into a limited Insights view that shows usage over time, as well
+as usage broken down by platform.
+
+### The New `expo-insights` Module
```
```suggestion
### Using the `expo-insights` module
```

brentvatne commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,33 @@
+
+### The New `expo-insights` Module
+
+To get the most of EAS Insights functionality, we offer the `expo-insights`
+client module. Developers can add this module to their project and gain more
+precise usage metrics, as well as additional breakdowns by app store version.
+Right now the `expo-insights` module is limited to sending client events only
+pertaining to cold starts of the app, but in the future we will expand
+`expo-insights` to offer new types of events and payloads, to support more
+advanced Insights functionality.
```
```suggestion
Developers can add the `expo-insights` module to their project and gain more
precise usage metrics (than what can be provided by just aggregating update requests), as well as additional breakdowns by app store version.
Currently, the module is limited to sending client events only
pertaining to cold starts of the app, but in the future we will expand
`expo-insights` to offer new types of events and payloads, to support more
advanced functionality.
```

christopherwalter committed:
Update docs/pages/eas-insights/introduction.mdx

Co-authored-by: Brent Vatne <brentvatne@gmail.com>

christopherwalter committed:
Update docs/pages/eas-insights/introduction.mdx

Co-authored-by: Brent Vatne <brentvatne@gmail.com>

christopherwalter committed:
Update docs/pages/eas-insights/introduction.mdx

Co-authored-by: Brent Vatne <brentvatne@gmail.com>

christopherwalter committed:
Update docs/pages/eas-insights/introduction.mdx

Co-authored-by: Brent Vatne <brentvatne@gmail.com>

Simek reviewed (commented):
Would be nice to add this page (and maybe EAS Insights section) somewhere in the sidebars, so it's accessible not only by URL.

christopherwalter committed:
Requested changes

christopherwalter commented:
@Simek agreed, though i don't think this is blocking for now. Can you elaborate re:how i can add this to the sidebar?

amandeepmittal commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,30 @@
+
+### Using the `expo-insights` module
+
+Developers can add the `expo-insights` module to their project and gain more
```
```suggestion
Developers can add the `expo-insights` library to their projects and gain more
```

amandeepmittal commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,30 @@
+### Using the `expo-insights` module
+
+Developers can add the `expo-insights` module to their project and gain more
+precise usage metrics (than what can be provided by just aggregating update requests), as well as additional breakdowns by app store version.
```
```suggestion
precise usage metrics (than those provided by aggregating update requests) and additional breakdowns by app store version.
```

amandeepmittal commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,30 @@
+Currently, the module is limited to sending client events only
+pertaining to cold starts of the app, but in the future we will expand
+`expo-insights` to offer new types of events and payloads, to support more
+advanced functionality.
```
```suggestion
Currently, the module is limited to sending client events only
of cold starts of the app. However, in the future, we will expand
`expo-insights` to offer new types of events and payloads to support more
advanced functionality.
```

christopherwalter replied:
Not going to change this one, IMO the original language is preferable

Simek commented:
Sure, you need to edit `navigation.js` file, and place the new page entry somewhere in the structure:
* https://github.com/expo/expo/blob/main/docs/constants/navigation.js

The "general" sidebar (Guides), below the "EAS Metadata" section looks like a possible place for the Insights docs, but feel free to place them where you deem they fit best. 🙂

amandeepmittal commented:
I'm not sure why some of my suggestions are marked outdated by GitHub.

christopherwalter committed:
PR fixes + add insights to sidebar

christopherwalter commented:
@amandeepmittal I incorporated your suggestions, take a look and let me know if everything looks alright. I also added Insights to the sidebar (I think). @Simek let me know if I did this correctly.

amandeepmittal commented on docs/pages/eas-insights/introduction.mdx:
```diff
@@ -0,0 +1,18 @@
+
+## Using the `expo-insights` module
+
+Developers can add the `expo-insights` library to their projects and gain more precise usage metrics (than those provided by just aggregating update requests) and additional breakdowns by app store version.  Currently, the module is limited to sending client events only pertaining to cold starts of the app, but in the future we will expand `expo-insights` to offer new types of events and payloads ...
```
```suggestion
Developers can add the `expo-insights` library to their projects and gain more precise usage metrics (than those provided by just aggregating update requests) and additional breakdowns by app store version. Currently, the module is limited to sending client events only pertaining to cold starts of the app, but in the future we will expand `expo-insights` to offer new types of events and payloads, to support more advanced functionality.
```

Small nit: extra whitespace.

amandeepmittal reviewed (approved):
Thank you for incorporating those changes and writing this doc @christopherwalter 🙏 

The reason CI is failing is because of the following listing issue:

```
  246:32  warning  Replace `⏎····makePage('eas-insights/introduction.mdx'),⏎··` with `makePage('eas-insights/introduction.mdx')`  prettier/prettier
  ```
  
  You'll have to remove that extra space.

Simek reviewed (approved):
Thank you for the updates, LGTM too! 👌

reichhartd commented on docs/pages/eas-insights/index.js:
```diff
@@ -0,0 +1,3 @@
+import redirect from '~/common/redirect';
+
+export default redirect('/eas-update/introduction');
```
This should be `insights` instead of `update`?
```suggestion
export default redirect('/eas-insights/introduction');
```

christopherwalter committed:
Update docs/pages/eas-insights/introduction.mdx

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

christopherwalter committed:
Fix link in index.js

christopherwalter committed:
Fix lint
